forte/genci/genci_vector_rdm.cc

```diff
@@ -125,7 +182,16 @@ std::shared_ptr<RDMs> GenCIVector::compute_rdms(GenCIVector& C_left, GenCIVector
             return std::make_shared<RDMsSpinDependent>(g1a, g1b, g2aa, g2ab, g2bb, g3aaa, g3aab,
                                                        g3abb, g3bbb);
         }
+        if (max_rdm_level == 4) {
+            return std::make_shared<RDMsSpinDependent>(g1a, g1b, g2aa, g2ab, g2bb, g3aaa, g3aab,
+                                                       g3abb, g3bbb, g4aaaa, g4aaab, g4aabb, g4abbb,
+                                                       g4bbbb);
+        }
     } else {
+        if (max_rdm_level == 4) {
+            throw std::runtime_error(
+                "Spin-free RDMs of order 4 or higher are not implemented in GenCISolver");
```

Comment:
This will take care if the user demands 4RDMs. We should still add the exceptions to the accessors of the SpinFreeRDMs class